src/Adapter/Product/Update/ProductTagUpdater.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use PrestaShop\PrestaShop\Adapter\Product\Repository\TagRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\LocalizedTags;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use Product;
+
+/**
+ * Methods related for product-tags update operations
+ */
+class ProductTagUpdater
+{
+    /**
+     * @var TagRepository
+     */
+    private $tagRepository;
+
+    /**
+     * @param TagRepository $tagRepository
+     */
+    public function __construct(
+        TagRepository $tagRepository
+    ) {
+        $this->tagRepository = $tagRepository;
+    }
+
+    /**
+     * Removes previous tags & sets new list of tags for a product.
+     *
+     * @param Product $product
+     * @param LocalizedTags[] $localizedTagsList
+     *
+     * @throws CannotUpdateProductException
+     * @throws CoreException
+     */
+    public function setProductTags(Product $product, array $localizedTagsList): void
+    {
+        $productId = new ProductId((int) $product->id);
+
+        // delete all tags for product if array is empty
+        if (empty($localizedTagsList)) {
+            $this->tagRepository->deleteTags($productId);
+
+            return;
+        }
+
+        foreach ($localizedTagsList as $localizedTags) {
```

Comment:
Maybe we should add a `PHPDoc` for `$localizedTags` type I know the array type is specified in the function doc, but it would avoid looking back to search the type and improve readability

Reply:
Im not a fan of many phpDocs, it will eventually be forgotten, outdated & misleading. I think it is enough that LocalizedTags has a typehint and it is an object, so its not too hard to see what it contains. (you don't even need to scroll up&down to see the typehint :smile: )

Reply:
IMO its enough :man_shrugging: 
If I add it even PHPstorm underlines it suggesting that its too much as it already inherits the typehint from method docblock.